src/main/java/com/google/sps/ColorUtils.java

```diff
@@ -0,0 +1,143 @@
+    colorList.add(new ColorName("Light Gray", 0xD3, 0xD3, 0xD3));
+    colorList.add(new ColorName("Light Green", 0x90, 0xEE, 0x90));
+    colorList.add(new ColorName("Light Pink", 0xFF, 0xB6, 0xC1));
+    colorList.add(new ColorName("LightYellow", 0xFF, 0xFF, 0xE0));
+    colorList.add(new ColorName("Lime", 0x00, 0xFF, 0x00));
+    colorList.add(new ColorName("Magenta", 0xFF, 0x00, 0xFF));
+    colorList.add(new ColorName("Maroon", 0x80, 0x00, 0x00));
+    colorList.add(new ColorName("Blue", 0x00, 0x00, 0xCD));       // MediumBlue
+    colorList.add(new ColorName("Purple", 0x93, 0x70, 0xDB));     // MediumPurple
+    colorList.add(new ColorName("Green", 0x00, 0xFA, 0x9A));      // MediumSpringGreen
+    colorList.add(new ColorName("Turquoise", 0x48, 0xD1, 0xCC));  // MediumTurquoise
+    colorList.add(new ColorName("Navy", 0x00, 0x00, 0x80));
+    colorList.add(new ColorName("Orange", 0xFF, 0xA5, 0x00));
+    colorList.add(new ColorName("Orange", 0xFF, 0x45, 0x00));     // OrangeRed
+    colorList.add(new ColorName("Pink", 0xFF, 0xC0, 0xCB));
+    colorList.add(new ColorName("Purple", 0x80, 0x00, 0x80));
+    colorList.add(new ColorName("Red", 0xFF, 0x00, 0x00));
+    colorList.add(new ColorName("Blue", 0x41, 0x69, 0xE1));       // RoyalBlue
+    colorList.add(new ColorName("Brown", 0x8B, 0x45, 0x13));      // SaddleBrown
+    colorList.add(new ColorName("Silver", 0xC0, 0xC0, 0xC0));
+    colorList.add(new ColorName("Blue", 0x87, 0xCE, 0xEB));       // SkyBlue
+    colorList.add(new ColorName("Turquoise", 0x40, 0xE0, 0xD0));
+    colorList.add(new ColorName("Violet", 0xEE, 0x82, 0xEE));
+    colorList.add(new ColorName("White", 0xFF, 0xFF, 0xFF));
+    colorList.add(new ColorName("Yellow", 0xFF, 0xFF, 0x00));
+    return colorList;
+  }
+  
+  /**
+   * Returns the color name of the closest color (to the query color provided as RGB).
+   * To find the closest color, gets the color list, and then computes the distance between 
+   * the query color and every color in the color list, based on their RGB values.
+   */
+  public String getColorNameFromRGB(int r, int g, int b) throws IllegalArgumentException {
+    if (!(0 < r && r < 255) || !(0 < g && g < 255) || !(0 < b && b < 255)) {
+      throw new IllegalArgumentException("RGB values outside [0, 255] range.");
+    } 
+
+    ArrayList<ColorName> colorList = initColorList();
+    
+    // Get the color name of the color with minimum RGB distance to the query color.
+    ColorName closestColorMatch = null;
+
+    int minRGBDistance = Integer.MAX_VALUE;
+    for (ColorName color : colorList) {
+      int currentRGBDistance = color.computeRGBDistance(r, g, b);
+
+      if (currentRGBDistance < minRGBDistance) {
+        minRGBDistance = currentRGBDistance;
+        closestColorMatch = color;
+      }
+    }
+    
+    if (closestColorMatch != null) {
+      return closestColorMatch.getName();
+    } else {
+      return NO_MATCHED_COLOR_MESSAGE;
```

Comment:
This will never happen, right? There will be at least one match.

Reply:
I agree.

If this were a possibility, you should prefer an exception over a custom string which could accidentally be interpreted as a color.